src/layout/stack.scss

```diff
@@ -0,0 +1,286 @@
+@media ($viewport-regular) {
+  .Stack--align-start-whenRegular {
+    align-items: flex-start;
+  }
+
+  .Stack--align-center-whenRegular {
+    align-items: center;
+  }
+
+  .Stack--align-end-whenRegular {
+    align-items: flex-end;
+  }
+
+  .Stack--align-baseline-whenRegular {
+    align-items: baseline;
+  }
+}
+
+@media ($viewport-narrow) {
+  .Stack--align-start-whenNarrow {
+    align-items: flex-start;
+  }
+
+  .Stack--align-center-whenNarrow {
+    align-items: center;
+  }
+
+  .Stack--align-end-whenNarrow {
+    align-items: flex-end;
+  }
+  .Stack--align-baseline-whenNarrow {
+    align-items: baseline;
+  }
+}
+
+/* AlignWrap */
+
+@media ($viewport-regular) {
+  .Stack--alignWrap-start-whenRegular {
+    align-content: flex-start;
+  }
+
+  .Stack--alignWrap-center-whenRegular {
+    align-content: center;
+  }
+
+  .Stack--alignWrap-end-whenRegular {
+    align-content: flex-end;
+  }
+  .Stack--alignWrap-distribute-whenRegular {
+    align-content: space-between;
+  }
+  .Stack--alignWrap-distributeEvenly-whenRegular {
+    align-content: space-evenly;
+  }
+}
+
+@media ($viewport-narrow) {
+  .Stack--alignWrap-start-whenNarrow {
+    align-content: flex-start;
+  }
+  
+  .Stack--alignWrap-center-whenNarrow {
+    align-content: center;
+  }
+  
+  .Stack--alignWrap-end-whenNarrow {
+    align-content: flex-end;
+  }
+  .Stack--alignWrap-distribute-whenNarrow {
+    align-content: space-between;
+  }
+  .Stack--alignWrap-distributeEvenly-whenNarrow {
+    align-content: space-evenly;
+  }
+}
+
+/* Spread */
+
+@media ($viewport-regular) {
+  .Stack--spread-start-whenRegular {
+    justify-content: flex-start;
+  }
+  .Stack--spread-center-whenRegular {
+    justify-content: center;
+  }
+  .Stack--spread-end-whenRegular {
+    justify-content: flex-end;
+  }
+  .Stack--spread-distribute-whenRegular {
+    justify-content: space-between;
+  }
+  .Stack--spread-distributeEvenly-whenRegular {
+    justify-content: space-evenly;
+  }
+}
+
+@media ($viewport-narrow) {
+  .Stack--spread-start-whenNarrow {
+    justify-content: flex-start;
+  }
+  .Stack--spread-center-whenNarrow {
+    justify-content: center;
+  }
+  .Stack--spread-end-whenNarrow {
+    justify-content: flex-end;
+  }
+  .Stack--spread-distribute-whenNarrow {
+    justify-content: space-between;
+  }
+  .Stack--spread-distributeEvenly-whenNarrow {
+    justify-content: space-evenly;
+  }
+}
+
+/* Wrap */
+
+@media ($viewport-regular) {
+  .Stack--wrap-whenRegular {
+    flex-wrap: wrap;
+  }
+
+  .Stack--nowrap-whenRegular {
+    flex-wrap: nowrap;
+  }
+}
+
+@media ($viewport-narrow) {
+  .Stack--wrap-whenNarrow {
+    flex-wrap: wrap;
+  }
+
+  .Stack--nowrap-whenNarrow {
+    flex-wrap: nowrap;
+  }
+}
+
+/* Divider */
+
+@media ($viewport-regular) {
+  .Stack--showDividers-whenRegular .Stack-divider {
+    display: block;
+  }
+}
+@media ($viewport-narrow) {
+  .Stack--showDividers-whenNarrow .Stack-divider {
+    display: block;
+  }
+}
+
+.Stack-divider {
```

Comment:
Instead of adding an `<hr />` with this class name, we could just do something like this:

```
.Stack > * + * {
  border-top: 1px solid var(--primer-border-default);
  padding-top: var(--primer-stack-gap-{size});
}
```

This selects all children except the first child and adds a top padding and border. See https://alistapart.com/article/axiomatic-css-and-lobotomized-owls/ for more info on patterns like this.

Reply:
@mperrotti I thought about that, but since Stack is a generic helper layout, we should assume instances need to support any elements inside. It could be a button, per instance, which already has `border` and `padding` properties defined in them. If we add that style, we'd be altering their renderings...

Reply:
Ah - great foresight! We can keep the `hr`s then